### DISCUSSION TITLE
Incorrect formatting in github changelog

### DISCUSSION THREAD

**Discussion by ayechantharsoe** [jreleaser/jreleaser#266] [Q&A]
### Task List

- [x] Steps to reproduce provided
- [ ] Stacktrace (if present) provided
- [ ] Example that reproduces the problem (link to git repository is ideal)
- [x] Full description of the issue provided (see below)

### Steps to Reproduce

1. Check for `trace.log` for any conflict line such as:
 ```
- fb3343d Merge remote-tracking branch 'origin/dev' into dev

# Conflicts:
#	src/main/resources/i18n/messages.properties
#	src/main/resources/i18n/messages_en.properties

 ```

### Expected Behaviour

Text size should be normal.

### Actual Behaviour

Tell us what happens instead
![image](https://user-images.githubusercontent.com/8254188/124363427-2d9af300-dc61-11eb-879f-4b1dd891bed4.png)

### Environment Information

- **Operating System**: Windows 10
- **JReleaser Version:** 0.5.0
- **JDK Version:** 11.0.9

Please paste stacktraces from `[out|build|target]/jreleaser/trace.log`.

**Comment by aalmiray** [comment]
What's the current state of your working copy? Does it have a merge conflict?

**Comment by ayechantharsoe** [comment]
Wherever there is merge conflict, I resolve the merge and use the default generated commit message during merge operation which seems like

```
# Conflicts:
#	src/main/resources/i18n/messages.properties
#	src/main/resources/i18n/messages_en.properties
```

**Comment by aalmiray** [comment]
As far as I can tell, lines starting with a `#` are treated as comments and should not be part of the commit message.
Here's the suggested text I get when creating a merge commit after resolving a merge conflict

```
Merge updates for readme

2nd row needed an update

# this is a comment
 # this is NOT a comment

# Conflicts:
#   readme.txt
#
# It looks like you may be committing a merge.
# If this is not correct, please run
#   git update-ref -d MERGE_HEAD
# and try again.


# Please enter the commit message for your changes. Lines starting
# with '#' will be ignored, and an empty message aborts the commit.
#
# On branch main
# All conflicts fixed but you are still merging.
#
# Changes to be committed:
#   modified:   readme.txt
#
```

Notice there are no spaces before the `#` signs except for a single line stating that's not a comment. When querying the full log I get:

```
commit fc82d300adffc83bdb4a6e794db4a66259417818 (HEAD -> main)
Merge: 330391d 663f88b
Author: Andres Almiray <aalmiray@gmail.com>
Date:   Sun Jul 4 11:30:49 2021 +0200

    Merge updates for readme
    
    2nd row needed an update
    
     # this is NOT a comment

commit 330391d739e889010f1d1718ba14d9b57e44d21f
Author: Andres Almiray <aalmiray@gmail.com>
Date:   Sun Jul 4 11:29:38 2021 +0200

    2nd commit

commit 663f88b48f0a75b5c0735008da758196a0ea1d47 (update)
Author: Andres Almiray <aalmiray@gmail.com>
Date:   Sun Jul 4 11:29:22 2021 +0200

    Fix readme

commit f81a8adab42387ec81641c1654005f0d78d92d4d
Author: Andres Almiray <aalmiray@gmail.com>
Date:   Sun Jul 4 11:29:03 2021 +0200

    1st commit
```

As you can appreciate, there's a line in the last commit message that is formatted as if it were a comment but it isn't. because there's a single space at the start of the line. Which makes me think that perhaps spaces/tabs are being added by your editor at the beginning of each line in the commit message.

If you were to inspect your git log you'll probably find commit messages that contain lines that should had been commented out. JReleaser reads the data as is from the Git log, thus if JReleaser shows lines that appear to be comments (but aren't) then those lines are effectively found in the commit message (when they probably shouldn't).

**Comment by aalmiray** [comment]
FWIW the following configuration

```yaml
release:
  gitlab:
    owner: aalmiray
```

Generates a changelog like this:

```
## Changelog

fc82d30 Merge updates for readme

330391d 2nd commit

663f88b Fix readme

f81a8ad 1st commit
```

Where as the following configuration:

```yaml
release:
  gitlab:
    owner: aalmiray
    changelog:
      formatted: ALWAYS
      change: '- {{commitShortHash}} {{commitTitle}}'
```

Generates a changelog like this

```
## Changelog

- fc82d30 Merge updates for readme

- 330391d 2nd commit

- 663f88b Fix readme

- f81a8ad 1st commit



## Contributors

Andres Almiray
```

Note that even commit `fc82d30` has multiple lines in its commit message only the first (or title) is taken into account and rendered in the final changelog. Relates to https://github.com/jreleaser/jreleaser/discussions/261
